api/timeseries_storage_api.go

```diff
@@ -21,6 +21,7 @@ import (
 )
 
 type TimeseriesStorageAPI interface {
+	AdjustTimerange(requested Timerange, slotLimit int) Timerange
```

Comment:
Two things. 1) Structurally I don't think this belongs on the TimeseriesStorageAPI. 2) I want to modify the name so it makes more sense.

If this belongs on the top-level TimeseriesStorage than it needs to be at least universal enough that would apply to a second use-case. How would this work if the backing storage was a single RRD file? What would "adjustment" mean in that case? If it doesn't generalize then I would make it an exported method on the blueflood type, OR we create a slightly higher-abstraction that would generalize.